cyflash/bootload.py

```diff
@@ -187,12 +189,24 @@ def write_rows(self, data):
 		for array_id, array in data.arrays.iteritems():
 			for row_number, row in array.iteritems():
 				i += 1
-				self.session.program_row(array_id, row_number, row.data)
-				actual_checksum = self.session.get_row_checksum(array_id, row_number)
-				if actual_checksum != row.checksum:
-					raise BootloaderError(
-						"Checksum does not match in array %d row %d. Expected %.2x, got %.2x! Aborting." % (
-							array_id, row_number, row.checksum, actual_checksum))
+                                if (data.checksum_type == 1):
+                                        self.session.program_row(array_id, row_number, row.data)
```

Comment:
You seem to have deleted the code that calculates checksums if the checksum type is CRC16, here.